extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerContext.java

```diff
@@ -83,6 +84,11 @@ public interface ControllerContext
    */
   TaskActionClient taskActionClient();
 
+  /**
+   * Task lock type.
+   */
+  TaskLockType taskLockType();
```

Comment:
It is needed to get the task lock type in `ControllerImpl`.
Since `ControllerImpl` doesn't have access to the task context (present in the `MSQControllerTask`), I had to add this.

Do you think there is a cleaner way to achieve the same?

Reply:
Do you think it is better to add a method to just return the task object itself?

```
MSQControllerTask task();
```

Reply:
This seems weird to me as well. `ControllerContext` seems like a place to add stuff that is specific for a server type (indexer/middle manager/ standalone server), as opposed to a general configuration that does not change across all the servers. What's the cyclic dependency if we move this out? 

> Do you think there is a cleaner way to achieve the same?

WorkerImpl has a reference to the WorkerTask during the construction time, which it uses to have access to the context. Can a similar approach be used?

Reply:
If WorkerImpl already does it, I guess it would make sense to just pass `MSQControllerTask` as a constructor arg to `ControllerImpl`. 

I looked at the code, it doesn't seem like there should be any cyclic dependency since `controller` is initialized in `MSQControllerTask.runTask()` and is not required on object creation.